src/irmin/object_graph.ml

```diff
@@ -173,11 +173,11 @@ module Make (Hash : HASH) (Branch : Type.S) = struct
           cache_size);
     let marks = Table.create cache_size in
     let mark key level = Table.add marks key level in
-    let todo = Stack.create () in
+    let todo (* ISSUE *) = Stack.create () in
```

Comment:
Treat order when `rev=true` (The default for Repo.iter and Private.Node.Graph.iter).
```
    #8 a(max)  b(max) #6
        \     /
      #7 c   d #5
          \ / \
        #4 e   f(min) #0
          / \
      #3 g   h #2
          \ /
           i(min) #1
```

Treat order when `rev=false`. (The default for Private.Commit.History.iter)
```
    #7 a(max)  b(max) #0
        \     /
      #8 c   d #1
          \ / \
        #3 e   f(min) #2
          / \
      #6 g   h #4
          \ /
           i(min) #5
```

The docstring states:
```
    [iter depth min max node edge skip rev ()] iterates in topological order
    over the closure graph starting with the `max` nodes
```

`rev=true` seems like a valid reverse tological order.

`rev=false` seems like a DFS ordering. Since the last node visited (c)
is nor a leaf nor a root, I don't think it qualifies as a topological order.

A treat order we might want when `rev=false`. (Inverse of `rev=true`)
```
    #0 a(max)  b(max) #2
        \     /
      #1 c   d #3
          \ / \
        #4 e   f(min) #8
          / \
      #5 g   h #6
          \ /
           i(min) #7
```

https://en.wikipedia.org/wiki/Topological_sorting#Depth-first_search might work but i'm unsure about the perfs and the behavior regarding `cache_size`.

Reply:
The order with `rev=false` and multiple `max` entries is indeed quite wrong 😱